File: visa-payment-go/internal/services/account_service.go
package services

import (
	"errors"
	"visa-payment-service/internal/dto"
	"visa-payment-service/internal/models"
	"visa-payment-service/internal/repository"
)

type AccountService struct {
	accountRepo *repository.AccountRepository
}

func NewAccountService(accountRepo *repository.AccountRepository) *AccountService {
	return &AccountService{
		accountRepo: accountRepo,
	}
}

func (s *AccountService) CreateAccount(req *dto.AccountRequest) (*dto.AccountResponse, error) {
	// Check if account with this document number already exists
	exists, err := s.accountRepo.ExistsByDocumentNumber(req.DocumentNumber)
	if err != nil {
		return nil, err
	}
	if exists {
		return nil, errors.New("account with this document number already exists")
	}

	account := &models.Account{
		DocumentNumber: req.DocumentNumber,
	}

	err = s.accountRepo.Create(account)
	if err != nil {
		return nil, err
	}

	return &dto.AccountResponse{
		AccountID:      account.ID,
		DocumentNumber: account.DocumentNumber,
	}, nil
}

func (s *AccountService) GetAccount(id uint) (*dto.AccountResponse, error) {
	account, err := s.accountRepo.GetByID(id)
	if err != nil {
		return nil, err
	}

	return &dto.AccountResponse{
		AccountID:      account.ID,
		DocumentNumber: account.DocumentNumber,
	}, nil
}

File: visa-payment-java/src/main/java/com/visa/paymentservice/service/AccountService.java
package com.visa.paymentservice.service;

import com.visa.paymentservice.dto.AccountRequest;
import com.visa.paymentservice.dto.AccountResponse;
import com.visa.paymentservice.model.Account;
import com.visa.paymentservice.repository.AccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

@Service
@Transactional
public class AccountService {
    
    @Autowired
    private AccountRepository accountRepository;
    
    public AccountResponse createAccount(AccountRequest request) {
        // Check if account with this document number already exists
        if (accountRepository.existsByDocumentNumber(request.getDocumentNumber())) {
            throw new IllegalArgumentException("Account with document number " + request.getDocumentNumber() + " already exists");
        }
        
        Account account = new Account(request.getDocumentNumber());
        Account savedAccount = accountRepository.save(account);
        
        return new AccountResponse(savedAccount.getAccountId(), savedAccount.getDocumentNumber());
    }
    
    @Transactional(readOnly = true)
    public Optional<AccountResponse> getAccount(Long accountId) {
        return accountRepository.findById(accountId)
                .map(account -> new AccountResponse(account.getAccountId(), account.getDocumentNumber()));
    }
    
    @Transactional(readOnly = true)
    public Optional<Account> findAccountById(Long accountId) {
        return accountRepository.findById(accountId);
    }
}
